tests/func/test_add.py

```diff
@@ -634,3 +634,20 @@ def test_should_protect_on_repeated_add(link, dvc_repo, repo_dir):
     dvc_repo.add(repo_dir.FOO)
 
     assert not os.access(repo_dir.FOO, os.W_OK)
+
+
```

Comment:
Hm, how about we skip that test explicitly like so
```
@pytest.mark.parametrize(
    "fname, ignored_fname",
    [
        ("file!with_weird#naming_[1].txt", r"/file\!with_weird\#naming_\[1\].txt"),
        pytest.param(
            "file!with*weird#naming_[1].t?t",
            r"/file\!with\*weird\#naming_\[1\].t\?t",
            marks=pytest.mark.skipif(os.name == "nt", reason="..."),
       ),
   ]
)
...
```
? This way we won't have `if os.name == "nt"` magic inside of the test function, so you could see that something is being skipped, instead of having to look into the test itself to find that magic `if`.

Reply:
Oh, wait. But actually, linux test is a full one, so we don't need windows test at all. The issue here is that you are trying to test it through the full `dvc add` stack, but it seems like you could really unit test the escaping itself (i.e. `_add_entry_to_gitignore`).